share/functions/__fish_print_hostnames.fish

```diff
@@ -108,7 +108,7 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
             read -z -l contents <$file
 
             # Print hosts from system wide ssh configuration file
-            string replace -rfi '^\s*Host\s+(\S.*?)\s*$' '$1' -- $contents | string split ' ' | string match -v '*\**'
+            string split '\n' -- $contents | string replace -rfi '^\s*Host\s+(\S.*?)\s*$' '$1' | string match -v '*\**'
             # Also extract known_host paths.
             set known_hosts $known_hosts (string replace -rfi '.*KnownHostsFile\s*' '' -- $contents)
```

Comment:
And this line also uses `$contents` so the same issue applies.

Reply:
`set known_hosts $known_hosts (string replace -rfi '.*KnownHostsFile\s*' '' -- $contents)` don't need remove quotes.
because
> If -r or --regex is given, PATTERN is interpreted as a Perl-compatible regular expression, and REPLACEMENT can contain C-style escape sequences like \t as well as references to capturing groups by number or name as $n or ${n}.

Reply:
It's not about the `'\n'` like above, it's the issue you were trying to fix with that - if there are multiple lines in $contents, the replacement (and filter, because of `-f`!) will still only be run once, because it's passed as one argument.

So this $contents also needs to be split.

Reply:
Oh, im stupid...
Let me using `read -z -d \n -la contents`...